File: Kotlin/Easy/Zamka/zamka.kt
fun main()
{
    var lowNumber: Long = readLine()!!.toLong()
    var highNumber = readLine()!!.toLong()
    var compNumber = readLine()!!.toLong()

    println(loopThroughNumbersLow(lowNumber, highNumber, compNumber))
    println(loopThroughNumbersHigh(highNumber, lowNumber, compNumber))
}

fun loopThroughNumbersHigh(_start: Long, _end: Long, comp: Long): Long
{
    for (i in _start downTo _end) {
        var answer: Long = CalcSumOfNumber(i);
        if(comp == answer) return i;
    }
    return 9999L
}

fun loopThroughNumbersLow(_start: Long, _end: Long, comp: Long): Long
{
    for (i in _start .. _end) {
        var answer: Long = CalcSumOfNumber(i);
        if(comp == answer) return i;
    }
    return 9999L
}

fun CalcSumOfNumber(_number: Long): Long 
{
    var sumNumber = 0L
    var number = _number
    while(number != 0L)
        {
            sumNumber += number % 10L;
            number /= 10L;
        }

        return sumNumber;
}

File: JavaScript/Easy/Zamka/zamka.js
const readLine = require("readline");

let inputNumbers = [];

const rl = readLine.createInterface({input: process.stdin, output:process.stdout});

var lowNumber, highNumber, compNumber;

rl.on('line', (line) =>
{
    inputNumbers.push(parseInt(line));
    if(inputNumbers.length == 3) rl.close();   
});

rl.on('close', (close) =>
{
    lowNumber = inputNumbers[0];
    highNumber = inputNumbers[1];
    compNumber = inputNumbers[2];
    console.log(loopThroughNumbersLow(lowNumber, highNumber, compNumber));
    console.log(loopThroughNumbersHigh(lowNumber, highNumber, compNumber));
});

 function loopThroughNumbersHigh( low,  high,  comp)
        {
            for (let i = high; i >= low; i--)
            {
                
                var answer = CalcSumOfNumber(i);
                //console.log(answer);
                if(comp == answer) return i;
            }
            return 10003;
        }

        function loopThroughNumbersLow( low,  high,  comp)
        {
            for (let i = low; i <= high; i++)
            {
                var answer = CalcSumOfNumber(i);
                if(comp == answer) return i;
            }
            return 10002;
        }

        function CalcSumOfNumber(number)
        {
            var sumNumber = 0;
            while(number != 0)
            {
                sumNumber += number % 10;
                number = Math.floor(number / 10);
            }
            
            return sumNumber;
        }